spotlight-client/src/metricsApi/utils.ts

```diff
@@ -77,12 +77,17 @@ export function recordIsProbation(record: ValuesType<RawMetricData>): boolean {
   return record.supervision_type === "PROBATION";
 }
 
-export type DemographicView =
-  | "total"
-  | "race"
-  | "gender"
-  | "age"
-  | NoFilterIdentifier;
+const DemographicViewList = [
+  "total",
+  "race",
+  "gender",
+  "age",
+  "nofilter",
+] as const;
+export type DemographicView = typeof DemographicViewList[number];
+export function isDemographicView(x: string): x is DemographicView {
+  return DemographicViewList.includes(x as never);
```

Comment:
Why is this type `never`? I thought `never` meant never having a value or never occurs, but but here it would be one of the demographic identifier strings. I'm confused.

Reply:
I ... am not totally sure? I think it's a type-safe way of saying "I don't care what the type of this thing is"? if I use `any` here ESLint complains and recommends I use `unknown` or `never` instead. `unknown` causes a TS compiler error but `never` doesn't? 

I agree with you that this doesn't actually make any sense though, maybe for readability I had better cast this to something more logical like `any` and deal with the consequences more explicitly.